src/Psalm/Internal/Type/SimpleAssertionReconciler.php

```diff
@@ -2951,6 +2955,58 @@ private static function reconcileClassConstant(
         return TypeCombiner::combine(array_values($matched_class_constant_types), $codebase);
     }
 
+    private static function reconcileValueOf(
+        Codebase $codebase,
+        TValueOf $assertion_type
+    ): ?Union {
+        $reconciled_types = [];
+
+        // For now, only enums are supported here
```

Comment:
Our docs say that `value-of<>` also works with arrays: https://psalm.dev/docs/annotating_code/type_syntax/utility_types/#value-oft

Reply:
Is there a test for this?

Reply:
btw: these changes do actually target assertions, even tho `value-of` worked with `array`, that is not the case for the assertion syntax.
I do not really have a need for this and thus, we should keep this targeting enums for now.

If some1 wants to add array assertions as well, the code can be extended.

Reply:
I was just making sure we don't loose any functionality.

Reply:
yah, I wasn't 100% sure as well. Since there was no test for the backed enum type as well, I just double checked that again. 😬